### PR TITLE
Add tests to experimental script-to-assets branching IO manager package

### PR DESCRIPTION
**Pull request by benpankow**
## Summary

Adds some basic unit tests to the script-to-assets branching IO manager experimental example. 

Restructures the example a bit to be a Python package to accommodate tests. Runs mypy/lint as well as simulating the branching behavior to verify that assets are materialized/read as expected.

## Test Plan

Unit tests.

### CODE REVIEW

**Comment by benpankow** [comment]
Current dependencies on/for this PR:
* master
  * **PR #11773** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/11773" target="_blank">Graphite</a> 
    * **PR #11645** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/11645" target="_blank">Graphite</a> 
      * **PR #11920** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/11920" target="_blank">Graphite</a>  👈
      * **PR #11785** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/11785" target="_blank">Graphite</a> 
      * **PR #11791** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/11791" target="_blank">Graphite</a> 

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/dagster-io/dagster/11920?utm_source=stack-comment).

**Comment by benpankow** [comment]
Current dependencies on/for this PR:
* master
  * **PR #11773** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/11773" target="_blank">Graphite</a> 
    * **PR #11645** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/11645" target="_blank">Graphite</a> 
      * **PR #11920** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/11920" target="_blank">Graphite</a>  👈
      * **PR #11785** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/11785" target="_blank">Graphite</a> 
      * **PR #11791** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/11791" target="_blank">Graphite</a> 

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/dagster-io/dagster/11920?utm_source=stack-comment).

**Review by schrockn — DISMISSED** [review]
rebase plz

File: examples/experimental/script_to_assets_branching_io_manager/script_to_assets_branching_io_manager_tests/test_branching_functionality.py [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+def _materialize_assets(
+    instance: DagsterInstance, workspace: WorkspaceRequestContext, asset_keys: List[AssetKey]
+) -> DagsterRun:
+    all_assets_job = next(
+        j
+        for j in workspace.get_repository_location("my_repo")
+        .get_repository(SINGLETON_REPOSITORY_NAME)
+        .get_all_external_jobs()
+        if j.name == ASSET_BASE_JOB_PREFIX
+    )
```
Comment (schrockn) [inline examples/experimental/script_to_assets_branching_io_manager/script_to_assets_branching_io_manager_tests/test_branching_functionality.py:103]: look at `get_implicit_job_def_for_assets` on RepositoryDefinition

File: examples/experimental/script_to_assets_branching_io_manager/script_to_assets_branching_io_manager_tests/test_branching_functionality.py [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+        script_to_assets_branching_io_manager.dev_defs.get_job_def(ASSET_BASE_JOB_PREFIX),
+        asset_selection=frozenset(asset_keys),
+        external_pipeline_origin=all_assets_job.get_external_origin(),
+    )
+
+    instance.launch_run(run.run_id, workspace)
+
+    poll_for_finished_run(instance, run.run_id, timeout=60)
+
+    return check.not_none(instance.get_run_by_id(run.run_id))
```
Comment (schrockn) [inline examples/experimental/script_to_assets_branching_io_manager/script_to_assets_branching_io_manager_tests/test_branching_functionality.py:114]: wait is this test going to download from hackernews API on every run? We shouldn't do that.

**Review by schrockn — CHANGES_REQUESTED** [review]
Looking like we are going a full download on every test? That is a bad idea. We could cache a response and read it off of disk or something.

**Pull request closed without merging**